Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PowderDiffILLDetScanReduction.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import (absolute_import, division, print_function)
+
+from mantid.kernel import StringListValidator, Direction, FloatArrayProperty, FloatArrayOrderedPairsValidator
+from mantid.api import PythonAlgorithm, MultipleFileProperty, Progress, WorkspaceGroupProperty
+from mantid.simpleapi import *
+
+
+class PowderDiffILLDetScanReduction(PythonAlgorithm):
+    _progress = None
+
+    def category(self):
+        return "ILL\\Diffraction;Diffraction\\Reduction"
+
+    def summary(self):
+        return 'Performs powder diffraction data reduction for D2B and D20 (when doing a detector scan).'
+
+    def name(self):
+        return "PowderDiffILLDetScanReduction"
+
+    def validateInputs(self):
+        issues = dict()
+
+        height_range = self.getProperty('HeightRange').value
+        if len(height_range) > 2:
+            issues['HeightRange'] = 'HeightRange must contain a minimum and maximum only'
+
+        return issues
+
+    def PyInit(self):
+
+        self.declareProperty(MultipleFileProperty('Run', extensions=['nxs']),
+                             doc='File path of run(s).')
+
+        self.declareProperty(name='NormaliseTo',
+                             defaultValue='Monitor',
+                             validator=StringListValidator(['None', 'Monitor']),
+                             doc='Normalise to monitor, or skip normalisation.')
+
+        self.declareProperty(name='UseCalibratedData',
+                             defaultValue=True,
+                             doc='Whether or not to use the calibrated data in the NeXus files.')
+
+        self.declareProperty(name='Output2DTubes',
+                             defaultValue=False,
+                             doc='Output a 2D workspace of height along tube against tube scattering angle.')
+
+        self.declareProperty(name='Output2D',
+                             defaultValue=False,
+                             doc='Output a 2D workspace of height along tube against the real scattering angle.')
+
+        self.declareProperty(name='Output1D',
+                             defaultValue=True,
+                             doc='Output a 1D workspace with counts against scattering angle.')
+
+        self.declareProperty(FloatArrayProperty(name='HeightRange', values=[], validator=FloatArrayOrderedPairsValidator()),
+                             doc='A pair of values, comma separated, to give the minimum and maximum height range (in m). If not specified '
+                                 'the full height range is used.')
+
+        self.declareProperty(WorkspaceGroupProperty('OutputWorkspace', '',
+                                                    direction=Direction.Output),
+                             doc='Output workspace containing the reduced data.')
+
+    def PyExec(self):
+        input_workspaces = self._load()
+        instrument_name = input_workspaces[0].getInstrument().getName()
```

Comment:
I would also validate if the loaded workspaces are time-indexed. This [trick](https://github.com/mantidproject/mantid/blob/master/Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PowderDiffILLDetEffCorr.py#L164) will do, but I wonder if we should export at least `isScanned` method to python?

Reply:
Hmm, a good question, this will still work for non-time indexed data so I am tempted to skip this check. I don't know of any, but there could conceivably be a case for D2B where you want to run this where you do not have any time indexing, but still want to do some of the tube magic.